src/golibmc.go

```diff
@@ -921,14 +1325,69 @@ func (client *Client) Stats() (map[string](map[string]string), error) {
 
 // Quit will close the sockets to each memcached server
 func (client *Client) Quit() error {
-	client.lock()
-	defer client.unlock()
+	client.lk.Lock()
 
-	errCode := C.client_quit(client._imp)
-	C.client_destroy_broadcast_result(client._imp)
+	close(client.openerCh)
+	if client.cleanerCh != nil {
+		close(client.cleanerCh)
+	}
+	for _, cr := range client.connRequests {
+		close(cr)
+	}
+	client.closed = true
+	client.lk.Unlock()
+	for _, cn := range client.freeConns {
+		if err := cn.quit(); err != nil {
+			return err
+		}
+	}
+	client.lk.Lock()
+	client.freeConns = nil
+	client.lk.Unlock()
+
+	return nil
+}
 
-	if errCode == C.RET_CONN_POLL_ERR || errCode == C.RET_RECV_ERR {
+func (cn *conn) expired(timeout time.Duration) bool {
+	if timeout <= 0 {
+		return false
+	}
+	return cn.createdAt.Add(timeout).Before(time.Now())
+}
+
+func (cn *conn) quit() error {
+	cn.Lock()
+	defer cn.Unlock()
+	if cn.closed {
+		return errors.New("duplicate conn close")
+	}
+	cn.closed = true
+	errCode := C.client_quit(cn._imp)
+	C.client_destroy_broadcast_result(cn._imp)
+	err := networkError(errorMessage[errCode])
+	if isBadConnErr(err) {
```

Comment:
isBadConnErr => isNetworkError ?

the meanings of "badConn" term that i used is different from badConnErr https://github.com/douban/libmc/pull/57/files#diff-b21c9a32b7563e26c49d7f215bc5a656R87 。

Reply:
okay, i see。 you want to put ErrBadConn together with all networkErrors，i am ok with this. ignore the above comment。